http_uniresolver/http_universal.py

```diff
@@ -31,17 +30,15 @@ def __init__(self):
 
     async def setup(self, _context: InjectionContext):
         """Preform setup, populate supported method list, configuration."""
-        config_file = os.environ.get(
-            "UNI_RESOLVER_CONFIG", Path(__file__).parent / "default_config.yml"
-        )
-        try:
-            with open(config_file) as input_yaml:
-                configuration = yaml.load(input_yaml, Loader=yaml.SafeLoader)
-        except FileNotFoundError as err:
-            raise ResolverError(
-                f"Failed to load configuration file for {self.__class__.__name__}"
-            ) from err
-        assert isinstance(configuration, dict)
+        plugin_conf = _context.settings.get("plugin_config", {}).get("http_uniresolver")
+        with open(
+            CONFIG_FILE,
+        ) as f:
+            # Default configuration
+            configuration = json.load(f)
+        if plugin_conf:
+            configuration.update(plugin_conf)
+
```

Comment:
I think the `deafult_config.json` should probably just be a statically defined dictionary in this (`http_universal.py`) file. I don't think we gain anything by loading it from a json file. Additionally, the default should be used only if no plugin config was found. In other words:

```suggestion
        # DEFAULT defined somewhere above
        configuration = _context.settings.get("plugin_config", {}).get("http_uniresolver")
        if not configuration:
            configuration = DEFAULT

```

Reply:
Probably you want to change the methods, but no the endpoint (by default). I guess merge with the changes is the better way to do it.